src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/ProvideHttpClientHandlerMaxResponseHeaderLengthValueCorrectly.cs

```diff
@@ -41,41 +42,57 @@ public override void Initialize(AnalysisContext context)
 
             context.RegisterCompilationStartAction(context =>
             {
-                context.RegisterOperationAction(context =>
-                {
-                    var propertyAssignment = (ISimpleAssignmentOperation)context.Operation;
-                    if (!IsHttpClientMaxResponseHeadersLengthAssignment(propertyAssignment))
-                    {
-                        return;
-                    }
-
-                    if (propertyAssignment.Value is null || !propertyAssignment.Value.ConstantValue.HasValue || propertyAssignment.Value.ConstantValue.Value is not int propertyValue)
-                    {
-                        return;
-                    }
-
-                    if (propertyValue > MaximumAlertLimit)
-                    {
-                        context.ReportDiagnostic(context.Operation.CreateDiagnostic(EnsureMaxResponseHeaderLengthRule, propertyValue));
-                    }
-                    
-                }, OperationKind.SimpleAssignment);
+                context.RegisterOperationAction(AnalyzeSimpleAssignmentOperationAndCreateDiagnostic, OperationKind.SimpleAssignment);
             });
         }
 
-        private static bool IsHttpClientMaxResponseHeadersLengthAssignment(ISimpleAssignmentOperation operation)
+        private void AnalyzeSimpleAssignmentOperationAndCreateDiagnostic(OperationAnalysisContext context)
+        {
+            var propertySymbol = GetMaxResponseHeaderLengthPropertySymbol(context);
+
+            if (propertySymbol is null)
+            {
+                return;
+            }
```

Comment:
This should be done in compilation start callback, not in the operation callback. The operation callback can capture the propertySymbol with `context.RegisterOperationAction(context => AnalyzeSimpleAssignmentOperationAndCreateDiagnostic(context, propertySymbol))`

Reply:
This one is also fixed.